src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -262,85 +263,105 @@ public IEnumerable<PackageItemViewModel> GetCurrent()
                 return Enumerable.Empty<PackageItemViewModel>();
             }
 
-            var listItemViewModels = new List<PackageItemViewModel>(capacity: _state.ItemsCount);
+            var listItemViewModels = new Dictionary<string, PackageItemViewModel>();
 
             foreach (PackageSearchMetadataContextInfo metadataContextInfo in _state.Results.PackageSearchItems)
             {
-                VersionRange allowedVersions = VersionRange.All;
-                VersionRange versionOverride = null;
+                var packageId = metadataContextInfo.Identity.Id;
+                var packageVersion = metadataContextInfo.Identity.Version;
+                var packageLevel = metadataContextInfo.TransitiveOrigins != null ? PackageLevel.Transitive : PackageLevel.TopLevel;
 
-                // get the allowed version range and pass it to package item view model to choose the latest version based on that
-                if (_packageReferences != null)
+                if (listItemViewModels.TryGetValue(packageId, out PackageItemViewModel existingListItem))
                 {
-                    IEnumerable<IPackageReferenceContextInfo> matchedPackageReferences = _packageReferences.Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.Identity.Id, metadataContextInfo.Identity.Id));
-                    VersionRange[] allowedVersionsRange = matchedPackageReferences.Select(r => r.AllowedVersions).Where(v => v != null).ToArray();
-                    VersionRange[] versionOverrides = matchedPackageReferences.Select(r => r.VersionOverride).Where(v => v != null).ToArray();
-
-                    if (allowedVersionsRange.Length > 0)
+                    if (packageLevel == PackageLevel.Transitive)
                     {
-                        allowedVersions = allowedVersionsRange[0];
+                        UpdateTransitiveInfo(existingListItem, metadataContextInfo);
                     }
 
-                    if (versionOverrides.Length > 0)
-                    {
-                        versionOverride = versionOverrides[0];
-                    }
+                    existingListItem.UpdateInstalledPackagesVulnerabilities(new PackageIdentity(packageId, packageVersion));
```

Comment:
@martinrrm Am I remembering correctly that we changed this to accept metadataContextInfo.Identity rather than instantiate a new PackageIdentity? I'm wondering how this made its way back here.

Reply:
I fixed this in the latest commit.